### PR TITLE
Additional support for --single-file option

### PR DESCRIPTION
**Pull request by nre-ableton**
This is related to the `--single-file` option, which was added in #213. This PR
updates the README and adds a new `checkSingleFile` function to the `groovylint`
singleton, and also bumps the version number so that we actually publish a new
Docker image and library release.

### CODE REVIEW

File: vars/groovylint.groovy [hunk 1]
```diff
@@ -66,3 +66,21 @@ void check(String includesPattern, Object groovylintImage = null, String extraAr
...
+void checkSingleFile(String filename) {
+  assert filename
+
+  String version = env['library.groovylint.version']
+  image = docker.image("abletonag/groovylint:${version}")
+  image.pull()
+  echo "Using groovylint Docker image: ${image.id}"
+
+  image.inside("-v ${env.WORKSPACE}:/ws") {
+    sh "python3 /opt/run_codenarc.py -- -includes=${includesPattern} ${extraArgs}"
```
Comment (ala-ableton) [inline vars/groovylint.groovy:84]: `includesPattern` is not defined.

File: vars/groovylint.groovy [hunk 2]
```diff
@@ -66,3 +66,21 @@ void check(String includesPattern, Object groovylintImage = null, String extraAr
...
   )
 }
+
+
+/**
+ * Check a single file with the {@code groovylint} Docker image.
+ * @param filename Filename to check.
+ */
+void checkSingleFile(String filename) {
+  assert filename
```
Comment (ala-ableton) [inline vars/groovylint.groovy:76]: Do we really have to pass a filename? Can't we pass a file path?

File: vars/groovylint.groovy [hunk 3]
```diff
@@ -66,3 +66,21 @@ void check(String includesPattern, Object groovylintImage = null, String extraAr
     groovylintImage: groovylintImage,
   )
 }
+
+
+/**
+ * Check a single file with the {@code groovylint} Docker image.
+ * @param filename Filename to check.
+ */
+void checkSingleFile(String filename) {
```
Comment (ala-ableton) [inline vars/groovylint.groovy:75]: Let's take the arguments with a `Map`, so we can easily extend them in the future,

File: vars/groovylint.groovy (hunk 1 above)
Comment (nre-ableton) [reply vars/groovylint.groovy]: Fixed in 3968e68.

File: vars/groovylint.groovy (hunk 2 above)
Comment (nre-ableton) [reply vars/groovylint.groovy]: Fixed in 3968e68.

File: vars/groovylint.groovy (hunk 3 above)
Comment (nre-ableton) [reply vars/groovylint.groovy]: Fixed in 3968e68.

File: vars/groovylint.groovy [hunk 4]
```diff
@@ -66,3 +66,38 @@ void check(String includesPattern, Object groovylintImage = null, String extraAr
...
+ *          </li>
+ *          <li>
+ *            {@code groovylintImage}: If specified, use this Docker image handle to run
+ *            {@code groovylint}. If {@code null}, then this function will try to fetch
+ *            {@code groovylint} from Docker hub using the same version number
+ *            corresponding to this library.
+ *          </li>
+ *        </ul>
+ */
+void checkSingleFile(Map args = [:]) {
```
Comment (ala-ableton) [inline vars/groovylint.groovy:86]: I guess we should also document this function on the `README`.

Comment (nre-ableton) [reply vars/groovylint.groovy:86]: Fixed in cadaa48.

**Review by ala-ableton — APPROVED** [review]
Once rebased/squashed, LGTM 👍🏻

**Comment by nre-ableton** [comment]
@ablbot autosquash

**Pull request merged**